### ISSUE TITLE
ViewAlias not working if the view hierarchy is complex

### ISSUE THREAD

**Issue by strangeliu** [nathantannar4/Engine#13]
ViewAlias always show the default body if the view hierarchy is complex. like..

```
struct SomeViewAlias: ViewAlias {
    
    var defaultBody: some View {
        Text("default body")
    }
}

SomeViewAlias()
...
// many many views and modifiers
...
.viewAlias(SomeViewAlias.self) {
    Circle()
}

```

Is it a bug related to Engine or a limitation of SwiftUI.

**Comment by nathantannar4** [comment]
Could you please share a complete example that repos?

**Comment by strangeliu** [comment]
After some tests, I found the ViewAlias is not working with AtomRoot View https://github.com/ra1028/swiftui-atom-properties,
I created a demo that can reproduce this issue https://github.com/strangeliu/ViewAliasIssue,

**Comment by nathantannar4** [comment]
Fixed in `1.8.7`, thanks for reporting!
